src/nrnpython/utils.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <cstddef>
+#include <stdexcept>
+
+#include <nanobind/nanobind.h>
+
+namespace nrn {
+namespace detail {
+
+template <typename T, size_t I>
+T get(const nanobind::tuple& t) {
+    if (I >= t.size()) {
+        throw std::runtime_error("Not enough arguments");
```

Comment:
I think we should let `t[I]` do its Python thing. Upon out-of-bounds it will set the Python exception and return Null, which is likely what we want 
https://docs.python.org/3/c-api/tuple.html#c.PyTuple_GetItem

Reply:
I think it's better if we are helpful with our error messages.  I think that my current exception isn't worded well, and could be better, but that's a detail I need to fill in later.

More importantly, without the check, the code will segfault.  Perhaps you have a different implementation in mind?

Reply:
So I guess something is confusing me. Why would `t[i]` segfault if it only calls `PyTuple_GetItem` under the hood?

Reply:
We now also have access to `libfmt`, so you can use that to improve the error message.

Reply:
I have updated the error message in the newest version.